lyse/utils.py

```diff
@@ -0,0 +1,121 @@
+#####################################################################
+#                                                                   #
+# /utils.py                                                      #
+#                                                                   #
+# Copyright 2013, Monash University                                 #
+#                                                                   #
+# This file is part of the program lyse, in the labscript suite     #
+# (see http://labscriptsuite.org), and is licensed under the        #
+# Simplified BSD License. See the license.txt file in the root of   #
+# the project for the full license.                                 #
+#                                                                   #
+#####################################################################
+
+import os
+import numpy as np
+
+# qt imports
+from qtutils import inmain_decorator
+from qtutils.qt import QtWidgets
+
+# labscript imports
+from labscript_utils.labconfig import LabConfig
+
+LYSE_DIR = os.path.dirname(os.path.realpath(__file__))
+
+# get port that lyse is using for communication
+try:
+    LABCONFIG = LabConfig(required_params={"ports": ["lyse"]})
+    LYSE_PORT = int(LABCONFIG.get('ports', 'lyse'))
+except Exception:
+    LABCONFIG = None
+    LYSE_PORT = 42519
```

Comment:
I'm not super comfortable with this. Instantiating a `LabConfig` just because you import `lyse.utils` has a bad code smell to it. It also isn't used by anything in this file. Seems like it was just move to fix some sort of circular dependency issue?

I'm sure there are a few possible solutions. One that may be the simplest is just to move the import of the labconfig inside the function that uses it, so that it can import it from `lyse`. That's perfectly acceptable to do, and a good way to break circular dependencies if one part only needs something at run time, not load time.